js/pricing.js

```diff
@@ -101,27 +101,39 @@ $(
         /**
          * Submits consent transaction.
          *
-         * <p>Prepares Privacy, Support Agreement consents data and transaction Url for sending and
-         * calls send transaction function.
+         * <p>Prepares transaction Url, Privacy and Support Agreement consents data, generates
+         * the reCaptcha token and adds it to the request data. After that calls send transaction
+         * function.
          */
         function submitOrder() {
+            const transactionUrl = `${apiUrl}/transaction`;
             const privacyConsent = $privacyConsent.prop("checked");
             const supportAgreementConsent = $supportAgreementConsent.prop("checked");
-            const data = JSON.stringify({privacyConsent, supportAgreementConsent});
-            const transactionUrl = `${apiUrl}/transaction`;
-            sendPaymentTransaction(transactionUrl, data);
+
+            showRedirect();
+
+            grecaptcha.ready(function() {
+                grecaptcha.execute("{{site.data.payment_config.reCaptchaSiteKey}}", {action: 'submitPaymentTransaction'}).then(function(token) {
```

Comment:
1) let's use arrow function.
2) please align code to make it readable. Here, I'd align `then` under `execute`.